PyDynamic/uncertainty/propagate_MonteCarlo.py

```diff
@@ -26,318 +26,320 @@
 __all__ = ["MC", "SMC"]
 
 class Normal_ZeroCorr():
-	"""
-	Multivariate normal distribution with zero correlation
-	"""
-	def __init__(self, mean=None, cov=None):
-		"""
-		Parameters
-		----------
-			loc: np.ndarray, optional
-				mean values, default is zero
-			scale: np.ndarray, optional
-				standard deviations for the elements in loc, default is zero
-		"""
-		if isinstance(mean, np.ndarray):
-			self.mean = mean
-			if isinstance(cov, np.ndarray):
-				assert (len(cov)==len(mean))
-				self.std = cov
-			elif isinstance(cov, float):
-				self.std = cov * np.ones_like(mean)
-			else:
-				self.std = np.zeros_like(mean)
-		elif isinstance(cov, np.ndarray):
-			self.std = cov
-			self.mean = np.zeros_like(cov)
-
-	def rvs(self, size=1):
-		# This function mimics the behavior of the scipy stats package
-		return np.tile(self.mean, (size, 1)) + np.random.randn(size, len(self.mean))*np.tile(self.std, (size, 1))
+    """
+    Multivariate normal distribution with zero correlation
+    """
+    def __init__(self, loc=None, scale=None):
+        """
+        Parameters
+        ----------
+            loc: np.ndarray, optional
+                mean values, default is zero
+            scale: np.ndarray, optional
+                standard deviations for the elements in loc, default is zero
+        """
+        if isinstance(loc, np.ndarray):
```

Comment:
We should reduce checks to either s
- isinstance log np.ndarry OR isinstance scale nd.array
- ifnboth ndarray, check length
-turn scalar intonvecotr of appropriate length